pkg/metrics/metrics.go

```diff
@@ -64,6 +66,11 @@ func NewMetrics() Metrics {
 			Name: "steward_pipelineruns_total",
 			Help: "total number of pipelineruns",
 		}),
+		State: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "steward_state_duration",
+			Help: "duration of states by pipelines",
+		},
```

Comment:
I am not sure if I understand this correctly.
Do you intent to use a pipeline run identifer as label? Then you will generate lot's of metrics with very few data points. That is not a good design.
Metrics should be pre-aggregated by the application emitting them.

Above, a histogram was used to report about pipeline run duration. Why not follow the same approach to report about wait times?